editor/src/components/editor/store/dispatch.tsx

```diff
@@ -510,9 +510,31 @@ export function editorDispatch(
   }
 
   const newVSCodeChanges = getVSCodeChanges(storedState.editor, frozenEditorState)
+  applyProjectChanges(frozenEditorState, newVSCodeChanges, updatedFromVSCode)
+
+  const shouldUpdatePreview =
+    anySendPreviewModel || frozenEditorState.projectContents !== storedState.editor.projectContents
+  if (shouldUpdatePreview) {
+    updateEmbeddedPreview(frozenEditorState.id, frozenEditorState.projectContents)
+  }
+
+  if (frozenEditorState.id != null && frozenEditorState.id != storedState.editor.id) {
+    storedState.workers.initWatchdogWorker(frozenEditorState.id)
+  }
+
+  return finalStore
+}
+
+function applyProjectChanges(
+  frozenEditorState: EditorState,
+  accumulatedChanges: AccumulatedVSCodeChanges,
```

Comment:
is `accumulatedChanges` a good name? this are the `getVSCodeChanges` for the current dispatch. does it mean accumulated changes of the current dispatch? 
I think the naming is backwards with `currentVSCodeChanges = combineAccumulatedVSCodeChanges(...` because `currentVSCodeChanges` is the _real_ accumulated changes waiting to be processed

Reply:
Yeah, I'd agree. I went with this name because of the name of the type, but I'm tempted to change that now...

Reply:
I believe the reasoning for the type's name btw was since it is an "accumulation" (in this case meaning "combination") of the file changes, highlight changes, and any selection change. But I agree it has become confusing.